lib/Cro/WebSocket/Client.pm6

```diff
@@ -54,12 +61,17 @@ class Cro::WebSocket::Client {
             my $magic = "258EAFA5-E914-47DA-95CA-C5AB0DC85B11";
             my $answer = encode-base64(sha1($key ~ $magic), :str);
 
-            my %options = headers => (Cro::HTTP::Header.new(name => 'Upgrade', value => 'websocket'),
-                Cro::HTTP::Header.new(name => 'Connection', value => 'Upgrade'),
-                Cro::HTTP::Header.new(name => 'Sec-WebSocket-Version', value => '13'),
-                Cro::HTTP::Header.new(name => 'Sec-WebSocket-Key', value => $key),
-                Cro::HTTP::Header.new(name => 'Sec-WebSocket-Protocol', value => 'echo-protocol'));
+            my @headers = do if self {
+                @!headers.clone.push: Cro::HTTP::Header.new(name => 'Sec-WebSocket-Key', value => $key)
```

Comment:
Cheaper and clearer code is is just `flat @!headers, Cro::HTTP::Header.new(name => 'Sec-WebSocket-Key', value => $key)`.